opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/StandardSpanDecoratorTest.java

```diff
@@ -0,0 +1,228 @@
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.matchesPattern;
+import static org.hamcrest.Matchers.not;
+import static org.hobsoft.hamcrest.compose.ComposeMatchers.hasFeature;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.DEFINED_PORT;
+import static org.springframework.http.HttpMethod.POST;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.HttpStatus.SERVICE_UNAVAILABLE;
+
+@ExtendWith(SpringExtension.class) // only needed for Spring 4 support
+@SpringBootTest(
+        webEnvironment = DEFINED_PORT,
+        properties = {
+                "opentracing.spring.web.client.enabled: false",
+                "spring.mvc.async.request-timeout: 1000"
+        })
+@EnableAutoConfiguration(excludeName = {
+        "org.springframework.boot.autoconfigure.web.ErrorMvcAutoConfiguration",
+        "org.springframework.boot.autoconfigure.web.servlet.error.ErrorMvcAutoConfiguration",
+})
+class StandardSpanDecoratorTest {
+
+    @Configuration
+    @Import(ExampleController.class)
+    static class TestConfiguration {
+
+        @Bean
+        public MockTracer tracer() {
+            return new MockTracer();
+        }
+
+        @Bean
+        @SuppressWarnings({"rawtypes", "unchecked"}) // generic as of Spring Boot 2
+        public FilterRegistrationBean tracingFilter(
+                final Tracer tracer,
+                final List<ServletFilterSpanDecorator> decorators) {
+
+            final TracingFilter filter = new TracingFilter(tracer, decorators, null);
+            final FilterRegistrationBean bean = new FilterRegistrationBean(filter);
+            bean.setAsyncSupported(true);
+
+            return bean;
+        }
+
+        @Bean
+        public ServletFilterSpanDecorator standardSpanDecorator() {
+            return new StandardSpanDecorator();
+        }
+
+        @Bean
+        public ServletFilterSpanDecorator httpUrlSpanDecorator() {
+            return new HttpUrlSpanDecorator();
+        }
+
+        @Bean
+        public ServletFilterSpanDecorator staticSpanDecorator() {
+            return new StaticSpanDecorator(singletonMap("test", "true"));
+        }
+
+    }
+
+    private final RestTemplate client = new RestTemplate();
+
+    public StandardSpanDecoratorTest() {
+        this.client.setErrorHandler(new DefaultResponseErrorHandler() {
+            @Override
+            public void handleError(final ClientHttpResponse response) {
+                // don't throw
+            }
+        });
+    }
+
+    @Autowired
+    private MockTracer tracer;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)